### PR TITLE
Update game messages create component to add post type

### PR DESCRIPTION
**Pull request by nazar**
References #127 - WIP

- [x] add migration to support characters and posting types in game message
- [x] update server schema and client queries for new `game_messages` structure
- [x] update game message rendering to render characters and message types
- [x] update create game message to support choosing a character or OOC post type

### CODE REVIEW

**Comment by mikelax** [comment]
Couple of issues found after testing:

1. Trying to update an OOC or IC post fails, the update ($input) is missing a value for `postType`.
Error:  Field value.postType of required type String! was not provided

2. When viewing the game messages as the GM, there are errors on the UI. 
One fix in `CreateGameMessage.js` function `CharacterProfile` is to use lodash to get the character.name. Even with this fix there are still other issues though. ie:
```js
function CharacterProfile(props) {
  const { character } = props;
  const characterUrl = _.get(character, 'profileImage.url');
  const characterName = _.get(character, 'name');
  const characterAvatar = characterUrl
    ? <Image avatar src={characterUrl}/>
    : <Icon name="user"/>;

  return (
    <React.Fragment>
      {characterAvatar} {characterName}
    </React.Fragment>
  );
}
```

File: client/src/components/Games/View/components/GameMessages.js [hunk 1]
```diff
@@ -128,9 +140,53 @@ class GameMessageContainerBase extends Component {
...
+        <Comment.Content>
+          <Comment.Author>{gameMessage.user.name}</Comment.Author>
+          <Comment.Metadata>
+            <div>
+              Posted {this._relativeDate(gameMessage.created_at)}
+            </div>
+            {this._lastEdited()}
+          </Comment.Metadata>
+          <Comment.Text>
+            <RichEditor message={gameMessage.message} ref={c => (this.editor = c)} readOnly={!(editing)} />
```
Comment (mikelax) [inline client/src/components/Games/View/components/GameMessages.js]: @nazar  should we refactor this `ref` to use the new react 16.3 `React.createRef()` 😮

File: client/src/components/Games/View/components/CreateMessage.js [hunk 2]
```diff
@@ -42,22 +62,132 @@ class CreateMessage extends Component {
...
+    const myGamePlayer = this._getMyGamePlayer();
+    return _.get(myGamePlayer, 'character');
   };
+
+}
+
+function CharacterProfile(props) {
+  const { character } = props;
+  const characterUrl = _.get(character, 'profileImage.url');
+  const characterAvatar = characterUrl
```
Comment (mikelax) [inline client/src/components/Games/View/components/CreateMessage.js:186]: should you add `const characterName = _.get(character, 'name');` and use `characterName` below on line 188?

File: client/src/components/Games/View/components/GameMessages.js (hunk 1 above)
Comment (nazar) [reply client/src/components/Games/View/components/GameMessages.js]: 🤔 - looking up `React.createRef()`

File: client/src/components/Games/View/components/CreateMessage.js (hunk 2 above)
Comment (nazar) [reply client/src/components/Games/View/components/CreateMessage.js:186]: Tempting but I think there will always be a `character.name` whereas `character,profileImage.url` is an optional field hence the `_.get` access.

**Review by mikelax — APPROVED** [review]
:shipit:

**Pull request merged**